### ISSUE TITLE
[WEB] Multiple Scrollbars: Gesture is not captured properly.

### ISSUE THREAD

**Issue by josne** [flutter/flutter#93677]
This issue might be related to [93674](https://github.com/flutter/flutter/issues/93674). As I am not sure, I create it separately. Let me know if I should combine them / close one.

## Steps to Reproduce

1. Execute `flutter run` on the code sample
2. move the vertical scrollbar.
3. The horizontal scrollbar disappears (as described in  [93674](https://github.com/flutter/flutter/issues/93674)).
4. Press and hold the vertical scrollbar and move the mouse horizontally. (It is necessary that you move it horizontally before moving it vertically)

**Expected results:** No scrollbar moves.

**Actual results:** The horizontal scrollbar moves.

![ScrollbarGesture](https://user-images.githubusercontent.com/42271940/141845178-b7123eb9-e6b9-45f7-aab0-3b003b53a51f.gif)

You can notice that this does not occur in [93674](https://github.com/flutter/flutter/issues/93674). The only difference in the sample code is the order of the Scrollbars in the widget tree. 

There are 2 possible workarounds that I found so far (both hacky and without deeper knowledge of the side effects): 
1. Change the order in the Widget tree. This still leads to the disappearance of the horizontal scrollbar (as described in [93674](https://github.com/flutter/flutter/issues/93674)). But the weird gesture mix up does not occur. 
2. Add the following code to the horizontal scrollbar. Not sure why this fixes it but maybe it helps understanding the issue.
```dart
notificationPredicate: (ScrollNotification notification) {
                  if (notification is ScrollUpdateNotification &&
                      notification.scrollDelta == null) return false;
                  return notification.depth == 0;
                },
```

<details>
<summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        body: MyWidget(),
      ),
    );
  }
}

class MyWidget extends StatefulWidget {
  @override
  _MyWidgetState createState() => _MyWidgetState();
}

class _MyWidgetState extends State<MyWidget> {
  ScrollController horizontalScrollController = ScrollController();
  ScrollController verticalScrollController = ScrollController();

  @override
  Widget build(BuildContext context) {
    return Center(
      child: SizedBox(
        width: 300,
        height: 500,
        child: Scrollbar(
          controller: horizontalScrollController,
          isAlwaysShown: true,
          notificationPredicate: (ScrollNotification notification) =>
              notification.depth == 0,
          child: Scrollbar(
            controller: verticalScrollController,
            isAlwaysShown: true,
            notificationPredicate: (ScrollNotification notification) =>
                notification.depth == 1,
            child: SingleChildScrollView(
              controller: horizontalScrollController,
              scrollDirection: Axis.horizontal,
              child: SizedBox(
                width: 500,
                child: ListView.builder(
                  controller: verticalScrollController,
                  itemCount: 50,
                  itemBuilder: (context, index) => Container(
                    width: 500,
                    height: 50,
                    color:
                        index.isEven ? Colors.greenAccent : Colors.blueAccent,
                    child: Text('Index: $index'),
                  ),
                ),
              ),
            ),
          ),
        ),
      ),
    );
  }
}

```

</details>

<details>
  <summary>Logs</summary>

```
$ flutter analyze
Analyzing bug...

   info - Use key in widget constructors - lib\main.dart:7:7 - use_key_in_widget_constructors
   info - Use key in widget constructors - lib\main.dart:18:7 - use_key_in_widget_constructors

2 issues found. (ran in 7.2s)

```

```
$ flutter doctor -v
[√] Flutter (Channel stable, 2.5.1, on Microsoft Windows [Version 10.0.19042.867], locale de-DE)
    • Flutter version 2.5.1 at C:\Users\***\tools\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision ffb2ecea52 (8 weeks ago), 2021-09-17 15:26:33 -0400
    • Engine revision b3af521a05
    • Dart version 2.14.2

[√] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at C:\Users\***\AppData\Local\Android\sdk
    • Platform android-30, build-tools 30.0.3
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Professional 2019 16.7.2)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional
    • Visual Studio Professional 2019 version 16.7.30413.136
    • Windows 10 SDK version 10.0.17763.0

[√] Android Studio (version 4.1)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[√] VS Code (version 1.62.1)
    • VS Code at C:\Users\***\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.28.0

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.19042.867]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 95.0.4638.54
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 95.0.1020.53

• No issues found!

```

</details>

**Comment by maheshmnj** [comment]
HI @josne, Thanks for filing the issue. I can reproduce the issue Dragging a vertical scrollbar horizontally simulates the Horizontal drag, I believe this should be solved by https://github.com/flutter/flutter/issues/87370

cc: @Piinks 

<details>
<summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel stable, 2.5.3, on macOS 12.0.1 21A559 darwin-arm, locale en-GB)
    • Flutter version 2.5.3 at /Users/mahesh/Documents/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 18116933e7 (5 weeks ago), 2021-10-15 10:46:35 -0700
    • Engine revision d3ea636dc5
    • Dart version 2.14.4

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/mahesh/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • ANDROID_HOME = /Users/mahesh/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 13.1, Build version 13A1030d
    • CocoaPods version 1.10.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)

[✓] IntelliJ IDEA Community Edition (version 2021.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 60.1.4
    • Dart plugin version 212.5080.8

[✓] VS Code (version 1.61.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.28.0

[✓] Connected device (3 available)
    • iPhone 12 (mobile) • 19EA5268-C931-4C3A-ACC6-71D3E849C450 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-5 (simulator)
    • macOS (desktop)    • macos                                • darwin-arm64   • macOS 12.0.1 21A559 darwin-arm
    • Chrome (web)       • chrome                               • web-javascript • Google Chrome 96.0.4664.45

• No issues found!
```

```
✓] Flutter (Channel master, 2.6.0-12.0.pre.719, on macOS 12.0.1 21A559 darwin-arm, locale en-GB)
    • Flutter version 2.6.0-12.0.pre.719 at /Users/mahesh/Documents/flutter_master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 34dbee66de (34 hours ago), 2021-11-14 18:46:02 -0500
    • Engine revision 140bd9399e
    • Dart version 2.16.0 (build 2.16.0-10.0.dev)
    • DevTools version 2.8.0

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/mahesh/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • ANDROID_HOME = /Users/mahesh/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.10.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)

[✓] IntelliJ IDEA Community Edition (version 2021.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 60.1.4
    • Dart plugin version 212.5080.8

[✓] VS Code (version 1.61.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.28.0

[✓] Connected device (3 available)
    • iPhone 12 (mobile) • 19EA5268-C931-4C3A-ACC6-71D3E849C450 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-5 (simulator)
    • macOS (desktop)    • macos                                • darwin-arm64   • macOS 12.0.1 21A559 darwin-arm
    • Chrome (web)       • chrome                               • web-javascript • Google Chrome 96.0.4664.45

• No issues found!
```
</details>

**Comment by Piinks** [comment]
I think this has been resolved by https://github.com/flutter/flutter/pull/87698

I checked this against the master branch and it did not reproduce, so it appears to have already been fixed. @maheshmnj it looks like you flutter doctor specifies 2.5, I have not seen this in 2.6 so I have removed the label.

```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel master, 2.6.0-12.0.pre.723, on macOS 11.6 20G165 darwin-x64, locale en)
[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0-rc2)
[!] Xcode - develop for iOS and macOS (Xcode 12.5.1)
    ! Flutter recommends a minimum Xcode version of 13.
      Download the latest version or update via the Mac App Store.
[✓] Chrome - develop for the web
[✓] Android Studio (version 2020.3)
[✓] IntelliJ IDEA Ultimate Edition (version 2021.1.3)
[✓] VS Code (version 1.62.2)
[✓] Connected device (2 available)

! Doctor found issues in 1 category.
```

I can see in the video that the horizontal scrollbar disappears when the bug happens, so I think this is https://github.com/flutter/flutter/issues/87697. Because of the notification predicate, the bug was causing both scrollbars to respond to scroll notifications in both axes. So in your repro, when you drag on the `vertical` scrollbar, it was actually the horizontal scrollbar painted on top of the vertical scrollbar, and dragging on it affected the horizontal axis. 😅 

If this is not fixed on the master branch, please let me know, otherwise we can close this as fixed!

**Comment by maheshmnj** [comment]
Thanks for taking a look. Closing as fixed on the master channel as of `master, 2.6.0-12.0.pre.719`, if anyone disagrees feel free to write in the comments and I will reopen it.


<details>
<summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel master, 2.6.0-12.0.pre.719, on macOS 12.0.1 21A559 darwin-arm, locale en-GB)
    • Flutter version 2.6.0-12.0.pre.719 at /Users/mahesh/Documents/flutter_master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 34dbee66de (2 days ago), 2021-11-14 18:46:02 -0500
    • Engine revision 140bd9399e
    • Dart version 2.16.0 (build 2.16.0-10.0.dev)
    • DevTools version 2.8.0

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/mahesh/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • ANDROID_HOME = /Users/mahesh/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.10.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)

[✓] IntelliJ IDEA Community Edition (version 2021.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 60.1.4
    • Dart plugin version 212.5080.8

[✓] VS Code (version 1.61.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.28.0

[✓] Connected device (3 available)
    • iPhone 12 (mobile) • 19EA5268-C931-4C3A-ACC6-71D3E849C450 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-5 (simulator)
    • macOS (desktop)    • macos                                • darwin-arm64   • macOS 12.0.1 21A559 darwin-arm
    • Chrome (web)       • chrome                               • web-javascript • Google Chrome 96.0.4664.45

• No issues found!
```
</details>

Thanks

**Comment by josne** [comment]
Thanks!
